### PR TITLE
Gimbal plugin

### PR DESCRIPTION
**Pull request by addy1997**
## Summary

A model plugin to control the `gimbal` added. This plugin has been adapted from the `gimbal controller` from this [repo](https://github.com/utiasSTARS/PhoenixDrone-sitl_gazebo/blob/master/include/gazebo_gimbal_controller_plugin.hh) and the `gimbal_small_2d` plugin from the `Gazebo (classic)`.

## Test it

* SDF file and meshes for the plugin can be found [here](https://github.com/addy1997/ign-gazebo/tree/gimbal-plugin/examples/plugin/gimbal_controller/gimbal). 

* In order to test the plugin, please use the gimbal model found [here](https://app.ignitionrobotics.org/OpenRobotics/fuel/models/Gimbal%20Small%202D).

* Broadly, the idea is to attach a gimbal to the base link of a drone and control it. I am still working on it and will add the files for the model on the fuel shortly.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]
Hi @addy1997, 

 - Can you explain why you are including the 3D files and the sdf inside the plugin folder. The right way to do it, it's to create a model in [ignition fuel](https://app.ignitionrobotics.org/dashboard), create an example world and fetch the model.
 - In general you should review the style, the indentation looks bad.
 - The year in the Licence should be 2021

Can you add some documentation about how to use it ?

**Comment by addy1997** [comment]
As the gimbal model already exists on fuel, I will remove the sdf file from
my commit, fix the indentation and add documentation.

**Comment by addy1997** [comment]
> Hi @addy1997,
> 
>     * Can you explain why you are including the 3D files and the sdf inside the plugin folder. The right way to do it, it's to create a model in [ignition fuel](https://app.ignitionrobotics.org/dashboard), create an example world and fetch the model.
> 
>     * In general you should review the style, the indentation looks bad.
> 
>     * The year in the Licence should be 2021
> 
> 
> Can you add some documentation about how to use it ?
@ahcorde,
1. I have fixed the indentation and reduced extra blank lines.
2. I have removed the mesh files and the model files.

File: examples/plugin/gimbal_controller/GimbalControllerPlugin.cc [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+          this->dataPtr->rollPub.Publish(m);
+
+          ss << yawJointPos;
+          m.set_data(ss.str());
+          this->dataPtr->yawPub.Publish(m);
+      }
+    }
+}
+////////////////////////////////////////////////////////////////////////////
+
```
Comment (ahcorde) [inline examples/plugin/gimbal_controller/GimbalControllerPlugin.cc:458]:
```suggestion

////////////////////////////////////////////////////////////////////////////
```

File: examples/plugin/gimbal_controller/GimbalControllerPlugin.hh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace systems
+{   
+    // Forward declaration
+    class GimbalControllerPluginPrivate;
+
+    /// \brief A plugin that simulates lift and drag.
+    class GimbalControllerPlugin
+
```
Comment (ahcorde) [inline examples/plugin/gimbal_controller/GimbalControllerPlugin.hh:37]:
```suggestion
```

File: examples/plugin/gimbal_controller/GimbalControllerPlugin.hh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                           EntityComponentManager &_ecm,
+                           EventManager &_eventMgr) override;
+
+        /// Documentation inherited
+        public: void PreUpdate(const UpdateInfo &_info,
+                           EntityComponentManager &_ecm) final;
+
+        /// \brief Private data pointer
+        private: std::unique_ptr<GimbalControllerPluginPrivate> dataPtr;
+
```
Comment (ahcorde) [inline examples/plugin/gimbal_controller/GimbalControllerPlugin.hh:61]:
```suggestion
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
Indentation is still wrong, In same places you are using tabs, in others 2 spaces, 4 spaces or other values.

Please use always spaces and use the same number of spaces for each indentation level, for example 2 spaces.

**Comment by addy1997** [comment]
> Indentation is still wrong, In same places you are using tabs, in others 2 spaces, 4 spaces or other values.
> 
> Please use always spaces and use the same number of spaces for each indentation level, for example 2 spaces.

@ahcorde, I hope this follows the guidelines.

**Review by ahcorde — CHANGES_REQUESTED** [review]
you should clean the history of the PR, you are including some commits that are not part of your contribution

**Comment by addy1997** [comment]
@ahcorde, I have pretty much badly messed up this branch. I will start from a fresh branch at some other time. Thanks and apologies for the inconvenience.

**Pull request closed without merging**